tests/tests-pending/rpc_pending_test.go

```diff
@@ -0,0 +1,347 @@
+	require.NoError(t, err)
+
+	// verify the pending trasnaction does not exist in the latest block info.
+	require.Empty(t, latestBlock)
+}
+
+func TestEth_Pending_GetTransactionByHash(t *testing.T) {
+	data := "0x608060405234801561001057600080fd5b5061011e806100206000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c806302eb691b14602d575b600080fd5b603360ab565b6040518080602001828103825283818151815260200191508051906020019080838360005b8381101560715780820151818401526020810190506058565b50505050905090810190601f168015609d5780820380516001836020036101000a031916815260200191505b509250505060405180910390f35b60606040518060400160405280600d81526020017f617261736b61776173686572650000000000000000000000000000000000000081525090509056fea264697066735822122060917c5c2fab8c058a17afa6d3c1d23a7883b918ea3c7157131ea5b396e1aa7564736f6c63430007050033"
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+	param[0]["data"] = data
+
+	txRes := util.Call(t, "eth_sendTransaction", param)
+	var txHash common.Hash
+	err := txHash.UnmarshalJSON(txRes.Result)
+	require.NoError(t, err)
+
+	rpcRes := util.Call(t, "eth_getTransactionByHash", []interface{}{txHash})
+	var pendingBlock map[string]interface{}
+	err = json.Unmarshal(rpcRes.Result, &pendingBlock)
+	require.NoError(t, err)
+
+	// verify the pending tx has all the correct fields from the tx sent.
+	require.NotEmpty(t, pendingBlock)
+	require.NotEmpty(t, pendingBlock["hash"])
+	require.Equal(t, pendingBlock["value"], "0xa")
+	require.Equal(t, pendingBlock["input"], data)
+}
+
+func TestEth_Pending_SendTransaction_PendingNonce(t *testing.T) {
+	currNonce := util.GetNonce(t, "latest")
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	// first transaction
+	_ = util.Call(t, "eth_sendTransaction", param)
+	pendingNonce1 := util.GetNonce(t, "pending")
+	require.Greater(t, uint64(pendingNonce1), uint64(currNonce))
+
+	// second transaction
+	param[0]["to"] = "0x7f0f463c4d57b1bd3e3b79051e6c5ab703e803d9"
+	_ = util.Call(t, "eth_sendTransaction", param)
+	pendingNonce2 := util.GetNonce(t, "pending")
+	require.Greater(t, uint64(pendingNonce2), uint64(currNonce))
+	require.Greater(t, uint64(pendingNonce2), uint64(pendingNonce1))
+
+	// third transaction
+	param[0]["to"] = "0x7fb24493808b3f10527e3e0870afeb8a953052d2"
+	_ = util.Call(t, "eth_sendTransaction", param)
+	pendingNonce3 := util.GetNonce(t, "pending")
+	require.Greater(t, uint64(pendingNonce3), uint64(currNonce))
+	require.Greater(t, uint64(pendingNonce3), uint64(pendingNonce2))
+}
+
+// func TestEth_Call_Pending(t *testing.T) {
```

Comment:
uncomment?

Reply:
we have not implmented any pending queries to `Call` because we do not have access to any contract pending states. will just remove!